diff -r b81b781ecafd execute_cmd.c
--- a/execute_cmd.c	Thu Aug 23 22:46:31 2012 -0400
+++ b/execute_cmd.c	Fri Aug 24 10:14:47 2012 -0400
@@ -2202,7 +2202,11 @@
   /* If the `lastpipe' option is set with shopt, and job control is not
      enabled, execute the last element of non-async pipelines in the
      current shell environment. */
-  if (lastpipe_opt && job_control == 0 && asynchronous == 0 && pipe_out == NO_PIPE && prev > 0)
+  if (lastpipe_opt
+#if defined (JOB_CONTROL)
+    && job_control == 0
+#endif
+    && asynchronous == 0 && pipe_out == NO_PIPE && prev > 0)
     {
       lstdin = move_to_high_fd (0, 0, 255);
       if (lstdin > 0)
diff -r b81b781ecafd lib/readline/util.c
--- a/lib/readline/util.c	Thu Aug 23 22:46:31 2012 -0400
+++ b/lib/readline/util.c	Fri Aug 24 10:14:47 2012 -0400
@@ -389,7 +389,7 @@
         break;
       s2++;
     }
-  while (--count != 0)
+  while (--count != 0);
 
   return (0);
 }
