diff -r 6bc67b489684 configure
--- a/configure	Thu Aug 23 19:40:19 2012 -0400
+++ b/configure	Fri Aug 24 10:14:34 2012 -0400
@@ -2007,7 +2007,7 @@
 	esac
 	;;
 
-    sunos*|solaris*)
+    sysv4*|sunos*|solaris*)
 	targos=sunos
 	case "$target_cpu" in
 	    i?86) targmach=i386 ;;
diff -r 6bc67b489684 configure.ac
--- a/configure.ac	Thu Aug 23 19:40:19 2012 -0400
+++ b/configure.ac	Fri Aug 24 10:14:34 2012 -0400
@@ -98,7 +98,7 @@
 	esac
 	;;
 
-    sunos*|solaris*)
+    sysv4*|sunos*|solaris*)
 	targos=sunos
 	case "$target_cpu" in
 	    i?86) targmach=i386 ;;
diff -r 6bc67b489684 libpcc/quad.h
--- a/libpcc/quad.h	Thu Aug 23 19:40:19 2012 -0400
+++ b/libpcc/quad.h	Fri Aug 24 10:14:34 2012 -0400
@@ -72,7 +72,7 @@
 typedef unsigned long long 	u_quad_t;
 #endif
 
-#ifdef __SunOS
+#if defined(__SunOS) || defined(__svr4__)
 typedef long long 		quad_t;
 typedef unsigned long long 	u_quad_t;
 #endif
